pythonforandroid/recipes/python2/__init__.py

```diff
@@ -56,89 +116,122 @@ def build_arch(self, arch):
+                info("Activate flags for ffi")
+                ffi_inc_dir = join(Recipe.get_recipe('libffi', self.ctx).get_build_dir(arch.arch), 'include')
+                ffi_libs_dir = Recipe.get_recipe('libffi', self.ctx).get_lib_dir(arch)
+                # info("\t-> Ffi Inc dir => {0}".format(ffi_inc_dir))
+                # info("\t-> Ffi Libs dir => {0}".format(ffi_libs_dir))
+
+                env['LIBFFI_CFLAGS'] = ' '.join([env['CFLAGS'], '-I{}'.format(ffi_inc_dir)])
+                env['LIBFFI_LIBS'] = ' '.join(['-L{}'.format(ffi_libs_dir), '-lffi'])
+
+                env['CFLAGS'] = ' '.join([env['CFLAGS'], '-I{}'.format(ffi_inc_dir)])
+                env['LDFLAGS'] = ' '.join([env['LDFLAGS'], '-L{}'.format(ffi_libs_dir), '-lffi'])
+
+            env['CFLAGS'] = ' '.join([env['CFLAGS'], '-Wformat'])
 
             configure = sh.Command('./configure')
             # AND: OFLAG isn't actually set, should it be?
             shprint(configure,
+                    'CROSS_COMPILE_TARGET=yes',
                     '--host={}'.format(env['HOSTARCH']),
                     '--build={}'.format(env['BUILDARCH']),
                     # 'OPT={}'.format(env['OFLAG']),
                     '--prefix={}'.format(realpath('./python-install')),
                     '--enable-shared',
-                    '--disable-toolbox-glue',
-                    '--disable-framework',
-                    _env=env)
-
-            # AND: tito left this comment in the original source. It's still true!
-            # FIXME, the first time, we got a error at:
-            # python$EXE ../../Tools/scripts/h2py.py -i '(u_long)' /usr/include/netinet/in.h
-        # /home/tito/code/python-for-android/build/python/Python-2.7.2/python: 1: Syntax error: word unexpected (expecting ")")
-            # because at this time, python is arm, not x86. even that, why /usr/include/netinet/in.h is used ?
-            # check if we can avoid this part.
-
-            make = sh.Command(env['MAKE'].split(' ')[0])
-            print('First install (expected to fail...')
-            try:
-                shprint(make, '-j5', 'install', 'HOSTPYTHON={}'.format(hostpython),
-                        'HOSTPGEN={}'.format(hostpgen),
-                        'CROSS_COMPILE_TARGET=yes',
-                        'INSTSONAME=libpython2.7.so',
-                        _env=env)
-            except sh.ErrorReturnCode_2:
-                print('First python2 make failed. This is expected, trying again.')
-
-
-            print('Second install (expected to work)')
-            shprint(sh.touch, 'python.exe', 'python')
-            shprint(make, '-j5', 'install', 'HOSTPYTHON={}'.format(hostpython),
-                    'HOSTPGEN={}'.format(hostpgen),
+                    '--with-system-ffi',
+                    '--disable-ipv6',
+                    # '--disable-toolbox-glue',
+                    # '--disable-framework',
+                    'ac_cv_file__dev_ptmx=no',
```

Comment:
Maybe setting as CPPFLAGS variable instead of passing as arguments into the configure script...what do you think?

PD: I Changed my source code with this approach but now I don't have time to test it, maybe tonight or tomorrow....